stdlib/functools.pyi

```diff
@@ -45,22 +46,32 @@ class _CacheInfo(NamedTuple):
     currsize: int
 
 @final
-class _lru_cache_wrapper(Generic[_T]):
-    __wrapped__: Callable[..., _T]
-    def __call__(self, *args: Hashable, **kwargs: Hashable) -> _T: ...
+class _lru_cache_wrapper(Generic[_P, _T]):
+    __wrapped__: Callable[_P, _T]
+    def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _T: ...
     def cache_info(self) -> _CacheInfo: ...
     def cache_clear(self) -> None: ...
-    def __copy__(self) -> _lru_cache_wrapper[_T]: ...
-    def __deepcopy__(self, __memo: Any) -> _lru_cache_wrapper[_T]: ...
+    def __copy__(self) -> _lru_cache_wrapper[_P, _T]: ...
+    def __deepcopy__(self, __memo: Any) -> _lru_cache_wrapper[_P, _T]: ...
+    if sys.version_info >= (3, 8):
+        @overload
+        def __get__(self, __instance: None, __owner: type[_S] | None = ...) -> _lru_cache_wrapper[_P, _T]: ...
+        @overload
+        def __get__(self, __instance: _S, __owner: type[_S] | None = ...) -> Callable[Concatenate[_S, _P], _T]: ...
```

Comment:
It seems you have this backward. I think it should be this:
```suggestion
        def __get__(self: _lru_cache_wrapper[Concatenate[_S, _Q], _T], __instance: _S, __owner: type[_S] | None = ...) -> Callable[_Q, _T]: ...
```
Where `_Q = ParamSpec("_Q")`. You'd also want to change the one for 3.8 below.